src_c/polygon.c

```diff
@@ -908,13 +908,69 @@ pg_polygon_rotate_ip(pgPolygonObject *self, PyObject *arg)
     Py_RETURN_NONE;
 }
 
+/*
+ * this function takes in `pgPolygonBase *` and
+ * it returns an int representing whether the polygon is convex or not
+ * (concave)
+ */
+static int
+_pg_polygon_is_convex_helper(pgPolygonBase *poly)
+{
+    Py_ssize_t i, i0, i1, i2;
+    Py_ssize_t verts_num = poly->verts_num;
+    Py_ssize_t count = 2 * verts_num;
+    double *vertices = poly->vertices;
+
+    /* A polygon is convex if and only if the cross products of all the
+     * adjacent edges are all of the same sign.
+     */
+    int sign = 0;
```

Comment:
The comment here could pe put before the variables and the sing variable closer to the others